app/lib/features/onboarding/widgets/encryption_backup_widget.dart

```diff
@@ -0,0 +1,147 @@
+    final availableApps = <Widget>[];
+
+    // Add copy button
+    availableApps.add(
+      _buildActionButton(
+        icon: Icons.copy,
+        onTap: () async {
+          await Clipboard.setData(ClipboardData(text: encryptionKey));
+          if (context.mounted) {
+            EasyLoading.showToast('Key copied to clipboard');
+          }
+        },
+        context: context,
+      ),
+    );
+
+    // Add share button
+    availableApps.add(
+      _buildActionButton(
+        icon: PhosphorIcons.share(),
+        onTap: () async {
+          await Share.share(encryptionKey);
+        },
+        context: context,
+      ),
+    );
+
+    void addButtonIfInstalled(
+      bool isInstalled,
+      IconData icon,
+      Future<void> Function() onTap,
+    ) {
+      if (isInstalled) {
+        availableApps.add(
+          _buildActionButton(
+            icon: icon,
+            onTap: () async {
+              await _buildShareContent();
+              if (!context.mounted) return;
+              await Future.delayed(const Duration(seconds: 2));
+              await onTap();
+            },
+            context: context,
+          ),
+        );
+      }
+    }
+
+    // 1Password (no platform restriction)
+    addButtonIfInstalled(
+      ref.watch(isAppInstalledProvider(ExternalApps.onePassword)).valueOrNull == true,
+      Icons.security,
+      () => shareToOnePassword(context: context),
+    );
+
+    // Bitwarden (no platform restriction)
+    addButtonIfInstalled(
+      ref.watch(isAppInstalledProvider(ExternalApps.bitwarden)).valueOrNull == true,
+      Icons.vpn_key,
+      () => shareToBitwarden(context: context),
+    );
+
+    // Keeper (iOS only)
+    addButtonIfInstalled(
+      Platform.isIOS &&
+          ref.watch(isAppInstalledProvider(ExternalApps.keeper)).valueOrNull == true,
+      Icons.lock_person,
+      () => shareToKeeper(context: context),
+    );
+
+    // LastPass (iOS only)
+    addButtonIfInstalled(
+      Platform.isIOS &&
+          ref.watch(isAppInstalledProvider(ExternalApps.lastPass)).valueOrNull == true,
+      Icons.password,
+      () => shareToLastPass(context: context),
+    );
+
+    // Enpass (iOS only)
+    addButtonIfInstalled(
+      Platform.isIOS &&
+          ref.watch(isAppInstalledProvider(ExternalApps.enpass)).valueOrNull == true,
+      PhosphorIcons.vault(),
+      () => shareToEnpass(context: context),
+    );
+
+    // ProtonPass (iOS only)
+    addButtonIfInstalled(
+      Platform.isIOS &&
+          ref.watch(isAppInstalledProvider(ExternalApps.protonPass)).valueOrNull == true,
+      Icons.shield,
+      () => shareToProtonPass(context: context),
+    );
+
+    return Wrap(
+      spacing: 16,
+      runSpacing: 16,
+      alignment: WrapAlignment.start,
+      children: availableApps,
+    );
+  }
+
+  Widget _buildActionButton({
+    required IconData icon,
+    required VoidCallback onTap,
+    required BuildContext context,
+  }) {
```

Comment:
to make it easier to use with above, we can just add:
```suggestion

  Widget _buildWithFirsCopyButton({
    required IconData icon,
    required VoidCallback onTap,
    required BuildContext context,
  } => _buildActionButton(icon: icon, context: context, onTap: () async {
              await _buildShareContent();
              if (!context.mounted) return;
              onTap();
  });

  Widget _buildActionButton({
    required IconData icon,
    required VoidCallback onTap,
    required BuildContext context,
  }) {
```

Reply:
Addressed by this [commit](https://github.com/acterglobal/a3/pull/2800/commits/4802d55146151a4e6ffd498dde85db1b805e0f4c)